pkgs/native_assets_cli/lib/src/api/build_config.dart

```diff
@@ -32,168 +46,173 @@ abstract class BuildConfig {
   /// dependency of another. For this it is convenient to know the packageRoot.
   Uri get packageRoot;
 
-  /// The target being compiled for.
-  ///
-  /// Not available in [dryRun].
-  Target get target;
-
   /// The architecture being compiled for.
   ///
-  /// Not available in [dryRun].
-  Architecture get targetArchitecture;
+  /// Not specified (`null`) during a [dryRun].
+  Architecture? get targetArchitecture;
 
   /// The operating system being compiled for.
-  OS get targetOs;
+  OS get targetOS;
 
   /// When compiling for iOS, whether to target device or simulator.
   ///
-  /// Required when [targetOs] equals [OS.iOS].
+  /// Required when [targetOS] equals [OS.iOS].
   ///
-  /// Not available in [dryRun].s
+  /// Not available during a [dryRun]. Will throw a [StateError] if accessed
+  /// during a [dryRun].
   IOSSdk? get targetIOSSdk;
```

Comment:
Style-guide correct capitalization would be `targetIosSdk`. I think.
I don't want to treat `iOS` as two words, one "i" and one "OS". But that would make `OS.iOS` incorrect, it too should be `OS.ios`. I can see how that breaks recognizability.
And making this `IOSSdk` is consistent with that.

So ... maybe keep it. Just not happy about it.

Reply:
We could do `OS.ios` since we're doing breaking changes anyway. However, that would make it inconsistent with https://api.dart.dev/stable/3.3.0/dart-io/Platform/isIOS.html

Let's keep it then.